sql/parser/normalize.go

```diff
@@ -371,9 +377,14 @@ func isConst(expr Expr) bool {
 }
 
 func isVar(expr Expr) bool {
-	switch expr.(type) {
+	switch t := expr.(type) {
 	case DReference, *ExistsExpr, *QualifiedName, *Subquery, ValArg:
 		return true
+	case *FuncExpr:
```

Comment:
Can you expand on why this is necessary? I understand the `isConstVisitor` check, but this one seems suspicious.